Tests/AuthenticationInterceptorTests.swift

```diff
@@ -50,47 +50,55 @@ class AuthenticationInterceptorTestCase: BaseTestCase {
         }
     }
 
-    enum OAuthError: Error {
+    enum TestAuthError: Error {
         case refreshNetworkFailure
     }
 
-    class OAuthAuthenticator: Authenticator {
+    final class TestAuthenticator: Authenticator {
         private(set) var applyCount = 0
         private(set) var refreshCount = 0
         private(set) var didRequestFailDueToAuthErrorCount = 0
         private(set) var isRequestAuthenticatedWithCredentialCount = 0
 
-        let refreshResult: Result<OAuthCredential, Error>?
+        let shouldRefreshAsynchronously: Bool
+        let refreshResult: Result<TestCredential, Error>?
         let lock = NSLock()
 
-        init(refreshResult: Result<OAuthCredential, Error>? = nil) {
+        init(shouldRefreshAsynchronously: Bool = true, refreshResult: Result<TestCredential, Error>? = nil) {
+            self.shouldRefreshAsynchronously = shouldRefreshAsynchronously
             self.refreshResult = refreshResult
         }
 
-        func apply(_ credential: OAuthCredential, to urlRequest: inout URLRequest) {
+        func apply(_ credential: TestCredential, to urlRequest: inout URLRequest) {
             lock.lock(); defer { lock.unlock() }
 
             applyCount += 1
 
             urlRequest.headers.add(.authorization(bearerToken: credential.accessToken))
         }
 
-        func refresh(_ credential: OAuthCredential,
+        func refresh(_ credential: TestCredential,
                      for session: Session,
-                     completion: @escaping (Result<OAuthCredential, Error>) -> Void) {
-            lock.lock(); defer { lock.unlock() }
+                     completion: @escaping (Result<TestCredential, Error>) -> Void) {
+            lock.lock()
 
             refreshCount += 1
 
-            let refreshResult = self.refreshResult ?? .success(
-                OAuthCredential(accessToken: "a\(refreshCount)",
-                                refreshToken: "a\(refreshCount)",
-                                userID: "u1",
-                                expiration: Date())
+            let result = refreshResult ?? .success(
+                TestCredential(accessToken: "a\(refreshCount)",
+                               refreshToken: "a\(refreshCount)",
+                               userID: "u1",
+                               expiration: Date())
             )
 
-            // The 100 ms delay here is important to allow multiple requests to queue up while refreshing
-            DispatchQueue.global(qos: .utility).asyncAfter(deadline: .now() + 0.1) { completion(refreshResult) }
+            if shouldRefreshAsynchronously {
+                // The 10 ms delay here is important to allow multiple requests to queue up while refreshing.
+                DispatchQueue.global(qos: .utility).asyncAfter(deadline: .now() + 0.01) { completion(result) }
+                lock.unlock()
+            } else {
+                lock.unlock()
+                completion(result)
+            }
```

Comment:
Seems like a solid addition. And I'm all for dropping down to 10 ms as long as CI can handle it.